### PR TITLE
Support for websocket auth, subaccounts, subscription for orders and fills

### PR DESCRIPTION
**Pull request by trading-peter**
Hi,

I added some stuff to this lib. What do you think?
One thing I couldn't figure out yet is how to make the websocket connection detect and reconnect if the internet connection goes away for a short time. I'm testing this by enabling and disabling my privacy vpn but it doesn't recognize the connection loss and continues to sind ping messages without pong messages ever arriving.
Any pointers what would be to do there?

### CODE REVIEW

**Comment by trading-peter** [comment]
I figured it out. Will update this PR tomorrow.

File: client.go [hunk 1]
```diff
@@ -35,17 +36,24 @@ func WithHTTPClient(client *http.Client) Option {
 	}
 }
 
-func WithAuth(key, secret string) Option {
+func WithAuth(key, secret string, subAccount ...string) Option {
```
Comment (grishinsana) [inline client.go:39]: If Client struct has c.subAccount field with string type, then no need to accept an array of strings.
```golang
func WithAuth(key, secret string, subAccount string) Option
```

File: websocket.go [hunk 2]
```diff
@@ -65,22 +80,22 @@ func (s *Stream) connect(requests ...models.WSRequest) (*websocket.Conn, error)
...
 	conn.SetPongHandler(func(msg string) error {
-		lastPong = time.Now()
-		if time.Now().Sub(lastPong) > websocketTimeout {
-			// TODO handle this case
-			s.printf("PONG response time has been exceeded")
-		} else {
-			s.printf("PONG")
-		}
+		s.printf("PONG")
+		conn.SetReadDeadline(time.Now().Add(s.wsTimeout))
```
Comment (grishinsana) [inline websocket.go:98]: what error will come after timeout?

**Comment by grishinsana** [comment]
> Hi,
> 
> I added some stuff to this lib. What do you think?
> One thing I couldn't figure out yet is how to make the websocket connection detect and reconnect if the internet connection goes away for a short time. I'm testing this by enabling and disabling my privacy vpn but it doesn't recognize the connection loss and continues to sind ping messages without pong messages ever arriving.
> Any pointers what would be to do there?

Connections will be reconnected if we get websocket.CloseGoingAway error in websocket.go:108

A couple of small comments. In general, everything looks good.

File: client.go (hunk 1 above)
Comment (trading-peter) [reply client.go:39]: But subAccount is optional. Isn't it award to force the user to set an empty string if he want's to connect to the main account? Is there a better pattern that I'm not aware of for optional arguments?

File: websocket.go (hunk 2 above)
Comment (trading-peter) [reply websocket.go:98]: Yes, that needs work. Will check soon-ish.

File: client.go (hunk 1 above)
Comment (grishinsana) [reply client.go:39]: I think you can add SetSubAccount method to Client and delete optional argument from here.
```golang
func (s *Stream) SetSubAccount(subAccount string) {
	s.mu.Lock()
	defer s.mu.Unlock()

	s. subAccount = subAccount
}```

Comment (trading-peter) [reply client.go:39]: :+1:

**Review by grishinsana — APPROVED** [review]

**Pull request merged**